src/datascience-ui/variable-view/variableViewPanel.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { handleLinkClick } from '../interactive-common/handlers';
+import { IMainWithVariables, IStore } from '../interactive-common/redux/store';
+import { IVariablePanelProps, VariablePanel } from '../interactive-common/variablePanel';
+import { actionCreators } from './redux/actions';
+
+// tslint:disable: no-suspicious-comment
+export type IVariableViewPanelProps = IMainWithVariables & typeof actionCreators;
+
+function mapStateToProps(state: IStore): IMainWithVariables {
+    return { ...state.main, variableState: state.variables };
+}
+
+// This is the top level UI element for our variable view panel, hosted in a vscode webviewView
+// It mimics the structure and state of InteractivePanel to be able to share creation / redux / actions
+// with the existing variable panels, but the UI contains only the Variable part of the UI
+export class VariableViewPanel extends React.Component<IVariableViewPanelProps> {
+    private renderCount: number = 0;
+
+    constructor(props: IVariableViewPanelProps) {
+        super(props);
+    }
+
+    public componentDidMount() {
+        document.addEventListener('click', this.linkClick, true);
+        this.props.editorLoaded(); // We don't have an editor, but this action signals that the view is ready, so just reuse it here
+    }
+
+    public componentWillUnmount() {
+        document.removeEventListener('click', this.linkClick);
+        this.props.editorUnmounted();
+    }
+
+    public render() {
+        const dynamicFont: React.CSSProperties = {
+            fontSize: this.props.font.size,
+            fontFamily: this.props.font.family
+        };
+
+        // If in test mode, update our count. Use this to determine how many renders a normal update takes.
+        if (this.props.testMode) {
+            this.renderCount = this.renderCount + 1;
+        }
+
+        // Return our variable panel, we wrap this in one more top level element "variable-view-main-panel" so that
+        // we can size and host it differently from the variable panel in the interactive window or native editor
+        return (
+            <div id="variable-view-main-panel" role="Main" style={dynamicFont}>
+                <button onClick={this.props.toggleVariableExplorer}>OPEN</button>
+                {this.renderVariablePanel(this.props.baseTheme)}
+            </div>
+        ); // NOTE: Currently the OPEN, button just exists to mimic the toggling of the variable view, make it easier to test when working
```

Comment:
The OPEN button is a slight hack that I want for now, and will remove later. Just an easy way for me to toggle on and off the variable explorer (in final version it will just always be open) as that triggers the variable fetching code to run. This is all behind an experiment still so customers won't see the view or anything in it.